### ISSUE TITLE
vscode issue with meta-shift-G (toggle source control)

### ISSUE THREAD

**Issue by fpigeonjr** [rbreaves/kinto#191]
**Describe the bug**
In microsoft visual studio code, to access **Source Control** the default is `ctrl-shift-G`.
When I try it with kinto it pulls up like a search window.
![image](https://user-images.githubusercontent.com/4629398/83255725-94083480-a176-11ea-800a-7c75ad8f944e.png)

**Expected behavior**
This keybinding should open up the Source Control tab like this.
![image](https://user-images.githubusercontent.com/4629398/83255867-d9c4fd00-a176-11ea-9082-f28fb4993265.png)

When I record the keybindings in vscode this is what it shows for `meta-shift-G`.
![image](https://user-images.githubusercontent.com/4629398/83256053-58219f00-a177-11ea-98e9-54d43985deae.png)

**Install Type:** Bare Metal 
**Distro:**
Operating System: Manjaro Linux 
KDE Plasma Version: 5.18.5
KDE Frameworks Version: 5.70.0
Qt Version: 5.14.2
Kernel Version: 5.6.12-1-MANJARO
OS Type: 64-bit
**Branch:** master

**Kinto Input Caret Status**
```
cat: /tmp/kinto/caret: No such file or directory

```

**Kinto Standard Log**
```
-- Logs begin at Thu 2020-05-28 14:26:10 CDT, end at Fri 2020-05-29 06:44:45 CDT. --
-- No entries --

```

**Additional context**
♥ Thanks for making this project, as a long time mac user it makes the transition to linux that much better.

**Comment by rbreaves** [comment]
This should be resolved now if you run back through the installer (you can install on top of your current install just fine).

The issue was that there are some select hotkeys I ported over from Sublime text so that I can use VS Code more easily and never spent the time to make it optional like I should have. That logic and prompting however has now been added (and has actually been part of the Windows install process already), so this update just bring the Linux side up to parity. Funny as it is usually the Windows part that lags behinds for various reasons lol.

**Comment by fpigeonjr** [comment]
:100: thanks for the quick fix on this @rbreaves!
I will try pulling in the changes and report back my findings. 

Yeah I live in all three OSs but Linux is the funnest and most challenging of them all in my opinion. 

I am also getting a weird error like this when I logout but will file a new issue for it.  Odd since everything is working fine.
![image](https://user-images.githubusercontent.com/4629398/83336120-8508ab80-a276-11ea-93e2-ebb2226b090b.png)

**Comment by fpigeonjr** [comment]
@rbreaves so I pulled in the updates and am still getting the issue.
Even though I am selecting `n` for both

- multi-language support 
- sublime text keymaps

They are selected anyways.

Selected no for both options

![2020-05-30_13-29](https://user-images.githubusercontent.com/4629398/83336571-3bba5b00-a27a-11ea-8128-a6554d4f323a.png)

:thinking: Says they have been selected?

![2020-05-30_13-30](https://user-images.githubusercontent.com/4629398/83336573-3f4de200-a27a-11ea-9c6d-4a856ecaf17c.png)

Please let me know if there what other troubleshooting steps I can take.

Thanks!

**Comment by rbreaves** [comment]
Well crap.. that is weird.. I had gone through it a few times to make sure it properly reading the boolean and yet it looks like it isn't if it is still enabling the remap.

**Comment by fpigeonjr** [comment]
:thinking: yeah odd, I even tried answering `yes` for **multi-language** and **sublime-text** and  got enabled for both.

So its enabling them regardless what option you choose.

**Comment by rbreaves** [comment]
@fpigeonjr Sorry about that, apparently I was not keeping track well of global variables vs local ones and that was the cause - why my quick test came back ok but the actual run kept failing. Should be able to use the latest master or dev with the proper changes now.

**Comment by fpigeonjr** [comment]
@rbreaves Thanks for the update. I pulled in `master` but am still experiencing the same issue.

In vscode, this is what I am getting for `ctrl-shift-G`

`shift+f3 ctrl+shift+`

Should I have to restart the PC?
I suppose I can try the `dev` branch as well...

**Comment by rbreaves** [comment]
Oh.. I don't think I properly labelled all of the Sublime key remaps for VS Code. I will step through all of the remaps and see where I have deviated.

**Comment by rbreaves** [comment]
> K("C-Shift-g"): K("Shift-f3"),              # find_prev

This appears to be correct, but that is really Cmd+Shift+G, so I'd think you're talking about the real Ctrl key but when I press that on my system, ctrl-shift-G, no toggling occurs - it just types in a capital G, that's it.

https://code.visualstudio.com/shortcuts/keyboard-shortcuts-macos.pdf

I would suggest that you post the contents of your ~/.config/kinto/kinto.py file for me to help you further, because I can't help you further without that and for the purpose of this conversation I will post the segment of code that matters and what I believe it should currently look like for you without Sublime remaps in VS Code.

```
# Keybindings for VS Code
define_keymap(re.compile(codeStr, re.IGNORECASE),{
    # Wordwise remaining - for VS Code
    # Alt-F19 hack fixes Alt menu activation
    K("M-Left"): [K("M-F19"),K("C-Left")],                  # Left of Word
    K("M-Right"): [K("M-F19"),K("C-Right")],                # Right of Word
    K("M-Shift-Left"): [K("M-F19"),K("C-Shift-Left")],      # Select Left of Word
    K("M-Shift-Right"): [K("M-F19"),K("C-Shift-Right")],    # Select Right of Word
    
    # K("C-PAGE_DOWN"): pass_through_key,         # cancel next_view
    # K("C-PAGE_UP"): pass_through_key,           # cancel prev_view
    K("C-M-Left"): K("C-PAGE_UP"),              # next_view
    K("C-M-Right"): K("C-PAGE_DOWN"),           # prev_view

    # VS Code Shortcuts
    K("C-g"): pass_through_key,                 # cancel Go to Line...
    K("Super-g"): K("C-g"),                     # Go to Line...
    K("F3"): pass_through_key,                  # cancel Find next
    K("C-h"): pass_through_key,                 # cancel replace
    K("C-M-f"): K("C-h"),                       # replace
    K("C-Shift-h"): pass_through_key,           # cancel replace_next
    K("C-M-e"): K("C-Shift-h"),                 # replace_next
    K("f3"): pass_through_key,                  # cancel find_next
    K("C-g"): K("f3"),                          # find_next
    K("Shift-f3"): pass_through_key,            # cancel find_prev
    K("C-Shift-g"): K("Shift-f3"),              # find_prev
    # K("Super-C-g"): K("C-f2"),                  # Default - Sublime - find_all_under
    # K("C-M-g"): K("C-f2"),                      # Chromebook - Sublime - find_all_under
    # K("Super-Shift-up"): K("M-Shift-up"),       # multi-cursor up - Sublime
    # K("Super-Shift-down"): K("M-Shift-down"),   # multi-cursor down - Sublime
    # K(""): pass_through_key,                    # cancel
    # K(""): K(""),                               #
}, "Code")
```

**Comment by fpigeonjr** [comment]
@rbreaves I am so sorry. You are absolutely correct. 

`alt-shift-g` on mac opens source control. I assumed that i would be the same binding on linux but that is not the case. On Linux, its `ctrl-shift-g` so in my head I was trying to use the kinto `meta-shift-g`.

Not sure if you want to include the keybinding into kinto but for now I am able to make it work with this in my `keybindings.json`.

```json

{
  "key": "alt+shift+g",
  "command": "workbench.view.scm"
 }

```

Sorry for my confusion and assumptions. Keep up the great support and best wishes with the project.

**Comment by rbreaves** [comment]
No worries, I needed to make that update any ways and it did take me a couple of tries to get it right. Only reason it had not already been done was because I had not came up with a good method for activating and deactivating code sections when I had first added a few sublime keymaps to VS Code. So all I am saying is I had intended to revisit this much earlier and forgot about it until you had made your post.

Also I can add the Alt+Shift+G keybind to map to the actual ctrl-shift-g combo just fine on Linux and Windows. I will close this ticket again once that is done.

**Comment by fpigeonjr** [comment]
> Also I can add the Alt+Shift+G keybind to map to the actual ctrl-shift-g combo just fine on Linux and Windows. I will close this ticket again once that is done.

That would be awesome and a great feature that at least I use everyday in vscode. Thanks again!

**Comment by rbreaves** [comment]
Added to dev branch, will be merged to master soon. Going ahead and closing this ticket though.
